src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,91 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.Set;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param userInput full user input string
+     * @return the command based on the user input
+     */
+    public static Command parse(String userInput) throws DukeException {
+        String commandWord = userInput.trim().split(" ", 2)[0];
+        String arguments;
+        Task task;
+        int taskNumber;
+
+        switch (commandWord) {
+
+            case AddCommand.ADD_TODO:
```

Comment:
Our guidelines says `case` should be on the same indentation as `switch`.

Reply:
You are right! Thanks for reminding me :)